docs/topics/data-analysis/data-analysis-connect-to-db.md

```diff
@@ -1,22 +1,167 @@
+   >
+   {type="tip"}
+
+3. Create a new Kotlin Notebook by selecting **File** | **New** | **Kotlin Notebook**.
+4. In the Kotlin Notebook, import the Kotlin DataFrame library by running the following command:
+
+```kotlin
+%use dataframe
+```
+
+5. Ensure you have access to an SQL database, such as MariaDB or MySQL.
+
+
+## Connect to database
+
+You can connect to and interact with an SQL database using specific functions from the [DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html). 
+You can use `DatabaseConfiguration` to establish a connection to your database and `getSchemaForAllSqlTables` to retrieve 
+the schema of all tables within it.
+
+Let's look at an example:
+
+1. **Import libraries:** Import the DataFrame library, which is essential for data manipulation tasks, along with
+the necessary Java libraries for SQL connectivity and utility functions: 
+
+```kotlin
+%use dataframe
+import java.sql.DriverManager
+import java.util.*
+```
+
+2. **Establish database connection:** Use `DatabaseConfiguration` to define your database's connection parameters, 
+including the URL, username, and password:
+
+```kotlin
+val URL = "YOUR_URL"
+val USER_NAME = "YOUR_USERNAME"
+val PASSWORD = "YOUR_PASSWORD"
+
+val dbConfig = DatabaseConfiguration(URL, USER_NAME, PASSWORD)
+```
+
+3. **Retrieve table schemas:** Once connected, use the `getSchemaForAllSqlTables()` function to fetch and display the 
+schema information for each table in the database:
+
+```kotlin
+val dataschemas = DataFrame.getSchemaForAllSqlTables(dbConfig)
+
+dataschemas.forEach { 
+    println("---Yet another table schema---")
+    println(it)
+    println()
+}
+```
+
+For more information on connecting to SQL databases, see [Read from SQL databases in the DataFrame documentation](https://kotlin.github.io/dataframe/readsqldatabases.html).
+
+## Retrieve and manipulate data
+
+After [establishing a connection to an SQL database](#connect-to-database), you can retrieve and manipulate data in Kotlin Notebook, utilizing the DataFrame library. 
+You can use `readSqlTable` to retrieve data. To manipulate data, you can use methods, such as [`.filter`](https://kotlin.github.io/dataframe/filter.html), [`.groupBy`](https://kotlin.github.io/dataframe/groupby.html), 
+and [`.convert`](https://kotlin.github.io/dataframe/convert.html). 
+
+Let's look at an example of connecting to an IMDB database and retrieving data about movies directed by Quentin Tarantino:
+
+1. **Fetch data from a table:** Use the `readSqlTable` function to retrieve data from the "movies" table, setting `limit` 
+to restrict the query to the first 100 records for efficiency:
+
+```kotlin
+val dfs = DataFrame.readSqlTable(dbConfig, tableName = "movies", limit = 100)
+```
+
+2. **Execute SQL query:**
+Use an SQL query to retrieve a specific dataset related to movies directed by Quentin Tarantino. 
+This query selects movie details and combines genres for each movie:
+
+```kotlin
```

Comment:
Adding code comments explaining the parts of the query would be helpful.

Reply:
Sure thing - I'm adding a comment!  👍 I intend to keep it as concise as possible as it's purely SQL and not DataFrame.